### PR TITLE
fix: portals held LP tokens in DeFi section

### PR DESCRIPTION
**Pull request by gomesalexandre**
## Description

Both a feat and a fix:

- Fixes the Zapper parsing of top-level held LP tokens in the DeFi section, which were broken twofolds:
  - wrong balance
  - wrong underlying assets
- Makes said LP assets clickable, routing to swapper

## Issue (if applicable)

- closes https://github.com/shapeshift/web/issues/7304

## Risk

> High Risk PRs Require 2 approvals

> What protocols, transaction types, wallets or contract interactions might be affected by this PR?

Low - confirm against Zapper held balances looks sane

## Testing

- Confirm LP assets as read-only opportunities are not read-only anymore (i.e if it's an LP asset, it should be clickable and route to swapper)
- Confirm underlying assets and staked amount looks sane against Zapper

### Engineering

- ^

### Operations

- [ ] :checkered_flag: My feature is behind a flag and doesn't require operations testing (yet)

- ^

## Screenshots (if applicable)

https://jam.dev/c/fce67eb9-71bf-43a7-9668-e7e870bcffe3

- Correct balance

Screenshot 2024-09-03 at 16 14 43
Screenshot 2024-09-03 at 16 14 56

- Corect underlying asset/s, tested against develop (right)

image

### CODE REVIEW

File: src/state/apis/zapper/zapperApi.ts [hunk 1]
```diff
@@ -368,7 +375,29 @@ export const zapper = createApi({
                   assets.map(asset => Object.assign(asset, { label })),
                 )
                 .map<ReadOnlyOpportunityType | undefined>(asset => {
+                  const chainId = zapperNetworkToChainId(asset.network)
+                  if (!chainId) throw new Error('chainIs is required')
```
Comment (NeOMakinG) [inline src/state/apis/zapper/zapperApi.ts:379]:
```suggestion
                  if (!chainId) throw new Error('chainId is required')
```

**Review by NeOMakinG — COMMENTED** [review]
small comment but looks sane overall, pending runtime testing

File: src/state/apis/zapper/zapperApi.ts (hunk 1 above)
Comment (gomesalexandre) [reply src/state/apis/zapper/zapperApi.ts:379]: fat fingers confirmed

**Review by NeOMakinG — APPROVED** [review]
## Develop
image
image


## This branch
image
image

## Product comments
The external link icon is now useless for these opportunities and unusable due to the swapper redirection click 
image

But looking good overall

**Comment by gomesalexandre** [comment]
Moving to draft so we don't get this in until its base PR is merged.

**Review by gomesalexandre — COMMENTED** [review]
Sanity retest before merge: https://jam.dev/c/be263f06-f717-469f-b1a8-996ed8724157

Note @NeOMakinG I've removed the read-only icon for LP assets as you've spotted, nice catch!

**Pull request merged**